packages/sitecore-jss-nextjs/src/editing/render-middleware.ts

```diff
@@ -16,16 +18,33 @@ export abstract class RenderMiddlewareBase {
     query: Partial<{ [key: string]: string | string[] }>
   ): { [key: string]: string } => {
     const params: { [key: string]: string } = {};
-    if (query[QUERY_PARAM_PROTECTION_BYPASS_SITECORE]) {
-      params[QUERY_PARAM_PROTECTION_BYPASS_SITECORE] = query[
-        QUERY_PARAM_PROTECTION_BYPASS_SITECORE
+    if (query[QUERY_PARAM_VERCEL_PROTECTION_BYPASS]) {
+      params[QUERY_PARAM_VERCEL_PROTECTION_BYPASS] = query[
+        QUERY_PARAM_VERCEL_PROTECTION_BYPASS
       ] as string;
     }
-    if (query[QUERY_PARAM_PROTECTION_BYPASS_VERCEL]) {
-      params[QUERY_PARAM_PROTECTION_BYPASS_VERCEL] = query[
-        QUERY_PARAM_PROTECTION_BYPASS_VERCEL
+    if (query[QUERY_PARAM_VERCEL_SET_BYPASS_COOKIE]) {
+      params[QUERY_PARAM_VERCEL_SET_BYPASS_COOKIE] = query[
+        QUERY_PARAM_VERCEL_SET_BYPASS_COOKIE
       ] as string;
     }
     return params;
   };
+
+  /**
+   * Get headers that should be passed along to subsequent requests
+   * @param {IncomingHttpHeaders} headers Incoming HTTP Headers
+   * @returns Object of approved headers
+   */
+  protected getHeadersForPropagation = (
+    headers: IncomingHttpHeaders
+  ): { [key: string]: string | string[] } => {
+    const result: { [key: string]: string | string[] } = {};
+    EDITING_PASS_THROUGH_HEADERS.forEach((header) => {
+      if (headers[header]) {
+        result[header] = headers[header]!;
```

Comment:
do we need the non-null assertion operator here? since we are already checking it.

Reply:
Yeah, without it I was still getting a TS error. Not quite sure why TS compile is having a problem in this particular case as you're right this usually isn't necessary if you're checking it.